### PR TITLE
work towards fixing an sdk upgrade issue

### PR DESCRIPTION
**Pull request by devoncarew**
One half of fixing an issue when upgrading the dart sdk in use by the flutter sdk (work towards https://github.com/flutter/flutter-intellij/issues/1155). This PR changes errors from the device daemon to not be hard errors - things that we show to users in dialogs.

@stevemessick

### CODE REVIEW

File: src/io/flutter/run/daemon/DeviceService.java [hunk 1]
```diff
@@ -205,10 +205,11 @@ private DeviceDaemon chooseNextDaemon(Refreshable.Request<DeviceDaemon> request)
...
     }
 
-    // Wait a bit to see if we get cancelled.
-    // This is to try to avoid starting a process only to immediately kill it.
+    // Wait a bit to see if we get cancelled. This is to try to avoid starting a process only to
+    // immediately kill it. Also, delay a bit in case the flutter tool just upgraded the sdk;
+    // we'll need a bit more time to start up.
     try {
-      Thread.sleep(50);
+      Thread.sleep(2000);
```
Comment (stevemessick) [inline src/io/flutter/run/daemon/DeviceService.java]: I don't understand why this timeout is increased. Is 2s enough to d/l a new SDK? Or is that some slack time to let the system stabilize after the SDK is installed?

Comment (devoncarew) [inline src/io/flutter/run/daemon/DeviceService.java]: It's a good question; I'm going to think more about the timing of the sdk upgrades (this delay is in here to work around some exceptions I saw if we run the `flutter` tool too soon after the new sdk is downloaded - possibly a race condition with the flutter cli tool?).

Comment (devoncarew) [inline src/io/flutter/run/daemon/DeviceService.java]: I think this is related to the time to snapshot flutter_tools. So:
- lock the flutter lock
- test to update the sdk
- build a snapshot of flutter_tools
- unlock the flutter lock
- run the flutter_tools dart code

However, it looks like we don't lock on the mac (https://github.com/flutter/flutter/blob/master/bin/flutter#L74), which is likely the cause of this issue.

Comment (devoncarew) [inline src/io/flutter/run/daemon/DeviceService.java]: (needing to introduce a delay)

**Pull request merged**